segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/GarbageCollector.java

```diff
@@ -374,22 +501,87 @@ public void report() {
         ChunkStorageMetrics.DYNAMIC_LOGGER.reportGaugeValue(SLTS_GC_QUEUE_SIZE, queueSize.get());
     }
 
-    @RequiredArgsConstructor
+    /**
+     * Represents a Task info.
+     */
+    public static abstract class AbstractTaskInfo {
+        public static final int DELETE_CHUNK = 1;
```

Comment:
Why do you use these hardcoded values for determining the types of tasks? What about if we add or delete operations in the future? I would suggest to use a subclass per task type and then discriminate tasks based on their instance types. Alternatively, enums could be also another approach here.

Reply:
Enums are clumsy and have lots of boiler plate and require me to do exactly same thing - define consts.

I feel subclass is overkill right now.